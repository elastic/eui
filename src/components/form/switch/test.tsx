import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
