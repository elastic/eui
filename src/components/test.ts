export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
