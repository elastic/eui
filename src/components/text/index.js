export {
  EuiText,
} from './text';

export {
  EuiTextColor,
} from './text_color';
