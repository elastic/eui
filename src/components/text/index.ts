export { EuiText, EuiTextProps } from './text';
export { EuiTextColor } from './text_color';
export { EuiTextAlign } from './text_align';
