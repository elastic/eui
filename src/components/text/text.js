import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {
  COLORS,
  EuiTextColor,
} from './text_color';

const textSizeToClassNameMap = {
  s: 'euiText--small',
  xs: 'euiText--extraSmall',
};

export const TEXT_SIZES = Object.keys(textSizeToClassNameMap);

export const EuiText = ({ size, color, children, className, ...rest }) => {

  const classes = classNames(
    'euiText',
    textSizeToClassNameMap[size],
    className
  );

  let optionallyColoredText;
  if (color) {
    optionallyColoredText = (
      <EuiTextColor color={color}>
        {children}
      </EuiTextColor>
    );
  } else {
    optionallyColoredText = children;
  }

  return (
    <div className={classes} {...rest}>
      {optionallyColoredText}
    </div>
  );
};

EuiText.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  size: PropTypes.oneOf(TEXT_SIZES),
  color: PropTypes.oneOf(COLORS),
};
