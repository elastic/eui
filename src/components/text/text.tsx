/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, HTMLAttributes } from 'react';
import classNames from 'classnames';
import { CommonProps, keysOf } from '../common';

import { TextColor, EuiTextColor } from './text_color';

import { EuiTextAlign, TextAlignment } from './text_align';

const textSizeToClassNameMap = {
  xs: 'euiText--extraSmall',
  s: 'euiText--small',
  m: 'euiText--medium',
};

export type TextSize = keyof typeof textSizeToClassNameMap;

export const TEXT_SIZES = keysOf(textSizeToClassNameMap);

export type EuiTextProps = CommonProps &
  Omit<HTMLAttributes<HTMLDivElement>, 'color'> & {
    textAlign?: TextAlignment;
    size?: TextSize;
    /**
     * **`secondary` color is DEPRECATED, use `success` instead**
     */
    color?: TextColor;
    grow?: boolean;
  };

export const EuiText: FunctionComponent<EuiTextProps> = ({
  size = 'm',
  color,
  grow = true,
  textAlign,
  children,
  className,
  ...rest
}) => {
  const classes = classNames(
    'euiText',
    textSizeToClassNameMap[size],
    className,
    {
      'euiText--constrainedWidth': !grow,
    }
  );

  let optionallyAlteredText;
  if (color) {
    optionallyAlteredText = (
      <EuiTextColor color={color} component="div">
        {children}
      </EuiTextColor>
    );
  }

  if (textAlign) {
    optionallyAlteredText = (
      <EuiTextAlign textAlign={textAlign}>
        {optionallyAlteredText || children}
      </EuiTextAlign>
    );
  }

  return (
    <div className={classes} {...rest}>
      {optionallyAlteredText || children}
    </div>
  );
};
