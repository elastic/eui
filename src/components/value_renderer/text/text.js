import { isNil } from 'lodash';

const createTextRenderer = (config = {}) => {
  const nilValue = config.nil || '';
  return (value) => isNil(value) ? nilValue : value.toString();
};

export const text = createTextRenderer();
text.with = (config) => createTextRenderer(config);
