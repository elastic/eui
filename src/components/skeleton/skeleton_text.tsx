/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, HTMLAttributes } from 'react';
import classNames from 'classnames';

import { CommonProps } from '../common';
import { useEuiTheme } from '../../services';
import { TextSize } from '../text/text';

import { EuiSkeletonLoading, _EuiSkeletonAriaProps } from './skeleton_loading';
import { euiSkeletonTextStyles } from './skeleton_text.styles';

export const LINES = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] as const;
export type LineRange = (typeof LINES)[number];

export type EuiSkeletonTextProps = CommonProps &
  HTMLAttributes<HTMLDivElement> &
  _EuiSkeletonAriaProps & {
    /**
     * Number of lines to display (between 1 to 10)
     */
    lines?: LineRange;
    /**
     * EuiText size to render
     */
    size?: TextSize;
  };

export const EuiSkeletonText: FunctionComponent<EuiSkeletonTextProps> = ({
  isLoading = true,
  lines = 3,
  size = 'm',
  className,
  contentAriaLabel,
  announceLoadingStatus,
  announceLoadedStatus,
  ariaLiveProps,
  ariaWrapperProps,
  children,
  ...rest
}) => {
  const euiTheme = useEuiTheme();
  const styles = euiSkeletonTextStyles(euiTheme);
  const lineCssStyles = [styles.euiSkeletonText, styles[size]];

  const lineElements = [];
  for (let i = 0; i < lines; i++) {
    lineElements.push(<span key={i} css={lineCssStyles} />);
  }

  return (
    <EuiSkeletonLoading
      isLoading={isLoading}
      loadingContent={
        <span className={classNames('euiSkeletonText', className)} {...rest}>
          {lineElements}
        </span>
      }
      loadedContent={children || ''}
      contentAriaLabel={contentAriaLabel}
      announceLoadingStatus={announceLoadingStatus}
      announceLoadedStatus={announceLoadedStatus}
      ariaLiveProps={ariaLiveProps}
      {...ariaWrapperProps}
    />
  );
};
