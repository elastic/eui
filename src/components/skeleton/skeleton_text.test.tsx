/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';
import { shouldRenderCustomStyles } from '../../test/internal';

import { TEXT_SIZES } from '../text/text';
import { EuiSkeletonText, LINES } from './skeleton_text';

describe('EuiSkeletonText', () => {
  shouldRenderCustomStyles(<EuiSkeletonText />);

  test('is rendered', () => {
    const { container } = render(<EuiSkeletonText {...requiredProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('lines', () => {
    LINES.forEach((count) => {
      test(String(count), () => {
        const { getByRole } = render(<EuiSkeletonText lines={count} />);
        const container = getByRole('progressbar');

        expect(container.querySelectorAll('span')).toHaveLength(count);
      });
    });
  });

  describe('size', () => {
    TEXT_SIZES.forEach((size) => {
      test(size, () => {
        const { container } = render(<EuiSkeletonText size={size} />);

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
