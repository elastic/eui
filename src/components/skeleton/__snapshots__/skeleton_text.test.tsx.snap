// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiSkeletonText is rendered 1`] = `
<div
  aria-busy="true"
  data-test-subj="euiSkeletonLoadingAriaWrapper"
>
  <span
    aria-label="Loading "
    class="euiSkeletonText testClass1 testClass2 emotion-euiTestCss"
    data-test-subj="test subject string"
    role="progressbar"
  >
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
  </span>
</div>
`;

exports[`EuiSkeletonText size m 1`] = `
<div
  aria-busy="true"
  data-test-subj="euiSkeletonLoadingAriaWrapper"
>
  <span
    aria-label="Loading "
    class="euiSkeletonText"
    role="progressbar"
  >
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-m"
    />
  </span>
</div>
`;

exports[`EuiSkeletonText size relative 1`] = `
<div
  aria-busy="true"
  data-test-subj="euiSkeletonLoadingAriaWrapper"
>
  <span
    aria-label="Loading "
    class="euiSkeletonText"
    role="progressbar"
  >
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-relative"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-relative"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-relative"
    />
  </span>
</div>
`;

exports[`EuiSkeletonText size s 1`] = `
<div
  aria-busy="true"
  data-test-subj="euiSkeletonLoadingAriaWrapper"
>
  <span
    aria-label="Loading "
    class="euiSkeletonText"
    role="progressbar"
  >
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-s"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-s"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-s"
    />
  </span>
</div>
`;

exports[`EuiSkeletonText size xs 1`] = `
<div
  aria-busy="true"
  data-test-subj="euiSkeletonLoadingAriaWrapper"
>
  <span
    aria-label="Loading "
    class="euiSkeletonText"
    role="progressbar"
  >
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-xs"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-xs"
    />
    <span
      class="emotion-euiSkeletonGradientAnimation-euiSkeletonText-xs"
    />
  </span>
</div>
`;
